src/deeprank_gnn/predict.py

```
@@ -139,10 +145,60 @@ def pdb_to_fasta(pdb_file_path: Path, main_fasta_fh: TextIOWrapper) -> None:
         main_fasta_fh.write(f">{root}.{chain.id}\n{sequence}\n")
 
 
+def calculate_checksum(file_path, algo="sha256") -> str:
+    h = hashlib.new(algo)
+    with open(file_path, "rb") as f:
+        for chunk in iter(lambda: f.read(8192), b""):
+            h.update(chunk)
+    return h.hexdigest()
+
+
+def download_weights(url: str, dest: str, timeout: int = 10) -> None:
+    try:
+        response = requests.get(url, stream=True, timeout=timeout)
+        response.raise_for_status()
+        with open(dest, "wb") as f:
+            for chunk in response.iter_content(chunk_size=8192):
+                if chunk:
+                    f.write(chunk)
+    except requests.RequestException:
+        pass
```

Here you need to handle the exceptions. According to the flow of the code, this path will be reached when the files were not found in disk and could not be downloaded.

As it is now it the `fetch_weights` will return a string that will point to a non existing. So you need to raise an error, because if these conditions are met the application cannot be executed.

Something like

```python
    try:
        # ...
    except requests.RequestException as e:
        raise RuntimeError(f"Failed to download weights from {url}: {e}")
    except Exception as e:
        raise RuntimeError(f"Unexpected error when trying to download the weights: {e})
```

---

updated